chore: add an AGENTS file (#4086)
jpivarski/awkward-1.0: AGENTS.md

# CLAUDE.md

> This file provides guidance to Claude Code (claude.ai/code) when working with code in this repository.

Reviewer: ```suggestion
# AGENTS.md

This file provides guidance to agents when working with code in this repository.
```

- Python-only changes need no rebuild (editable install). Rebuilding `awkward-cpp` is only needed when C++ changes; for fast iteration install build deps and use `pip install --no-build-isolation --check-build-dependencies ./awkward-cpp`.

## Testing

```bash
python -m pytest -n auto tests                      # main test suite
python -m pytest tests/test_XXXX-description.py     # single test file
python -m pytest -n auto awkward-cpp/tests-spec         # generated kernel-spec tests (Python reference impls)
python -m pytest -n auto awkward-cpp/tests-cpu-kernels  # generated tests against compiled kernels
```

CUDA suites (`tests-cuda`, `tests-cuda-kernels`) require an Nvidia GPU + CuPy. The generated kernel test dirs only exist after `nox -s prepare` (the `--tests` target).

Test files are named `tests/test_XXXX-description.py` where `XXXX` is the GitHub issue or PR number (enforced by `dev/validate-test-names.py`). New code needs a new test file following this convention.

Warnings are errors in pytest (`filterwarnings = ["error", ...]`).

## Linting

```bash

> pre-commit run -a    # or: nox -s lint    (user preference: prek -a --quiet)

Reviewer: Should we just fully switch to recommending `prek`?

Author: Yes, IMO :)

Author: Also, agents should use `--quiet`, it saves some context

Author: ```suggestion
prek -a --quiet    # or: nox -s lint (or pre-commit run -a if prek is missing)
```
